### PR TITLE
Drastically improved import speed (~10 times faster)

### LINKED ISSUE #121: Current implementation of progress bar causes lots of overhead during import
Incrementing progress during import is currently slowing down the application immensely, and needs to be reworked.

### PR DESCRIPTION
**Pull request by MikeStower**
By reworking the components used to track progress and run queries, the speed of importing data has been improved by an order of magnitude.

Resolves #121

### CODE REVIEW

File: Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public class ApsimXporter : ProgressTracker
         public override int Items => Experiments.Count();
 
         /// <inheritdoc/>
-        public override int Steps => Items * 28;
+        public override int Steps => Items * 29;
```
Comment (jbrider) [inline Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs:42]: what is 29?

File: Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs [hunk 2]
```diff
@@ -74,7 +74,7 @@ private async Task<Unit> InsertTable(DataTable table)
                     command = new InsertPlotsCommand() 
                     { 
                         Table = table,
-                        IncrementProgress = OnIncrementProgress
+                        Progress = Progress
```
Comment (jbrider) [inline Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs:77]: ??

Comment (MikeStower) [reply Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs:77]: ExcelImporter has a property called Progress. That object needs to be passed into the query to track progression, so InsertPlotCommand also has a property called Progress which references the original Progress through this assignment.

File: Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs (hunk 1 above)
Comment (MikeStower) [reply Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs:42]: A magic number that I haven't spent the time figuring out how to get rid of yet. The progress bar for exporting increments every time a new APSIM model is created and added to the output simulation. 29 is the number of models that are created per simulation, so the number of steps in the progress bar must be a multiple of 29 for the bar to fill.

At some point I plan to find a different way to track the output progress, I've just been prioritising other work.

File: Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs (hunk 2 above)
Comment (jbrider) [reply Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs:77]: one of these names has to change

File: Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs (hunk 1 above)
Comment (jbrider) [reply Infrastructure/Infrastructure/ProgressTrackers/ApsimXporter.cs:42]: var numModelsToExport = 29;

File: Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs (hunk 2 above)
Comment (jbrider) [reply Infrastructure/Infrastructure/ProgressTrackers/ExcelImporter.cs]: This is one of those github changes situations where it only shows me a small piece and 
I didn't take enough time work out what it was doing ie: passing down local properties to children - which is still ugly but it probably didn't need to be changed - I needed to take a look at the bigger picture. 

Weren't commands using DI - or does it need a specific instance of the progress tracker?

While changing the name of ProgressTracker to TaskRunner does make it sound more like an 'is-a' relationship - if all that it is doing is tracking progress then the original name is probably more suitable but should be changed to a 'has-a' relationship? What do you think?

**Pull request merged**